CONTRIBUTING.md

```diff
@@ -0,0 +1,101 @@
+3. Push the code changes from your local clone to your fork.
+4. Open a pull request from your fork's feature branch to GlotPress's `develop` branch. Pull request name should be `ISSUEID-keywords`. Take a look at [one such pull request](https://github.com/GlotPress/GlotPress-WP/pull/241) to get an idea of it.
+
+We use the [Git workflow](https://github.com/GlotPress/GlotPress-WP/wiki/5.-Git-workflow). Please have a read through, as it will make everyone's life easier. Should you have questions about that, reach out to us on Slack (see below).
+
+Pull request should have `Fixes #issue` / `Closes #issue`, or, if it's a partial solution, `Part of #issue` in the message to [enable automatic issue closing when a PR is merged](https://help.github.com/articles/closing-issues-via-commit-messages/).
+
+It doesn't matter if the code isn't perfect. The idea is to get it reviewed early and iterate on it.
+
+Lastly, please follow the [WordPress Coding Standards](http://make.wordpress.org/core/handbook/coding-standards/).
+
+## Testing patches / PRs
+
+1. Check out the branch the PR was submitted on.
+2. Leave feedback on the PR. Write up what you tested, what your results were and whether it solves the issue or not.
+
+If you need help with checking out the branch, or the PR was submitted from someone's fork, GitHub provides help in checking out the correct branch. Navigate to the PR in question, and you should see a link to command line instructions. Follow those instructions.
+
+![](https://cloud.githubusercontent.com/assets/617637/12953405/9df99808-d01a-11e5-88eb-958d23871f87.png)
+
+## Running and writing tests
+
+There are two types of automated tests:
+
+* unit tests, implemented using [PHPUnit](http://phpunit.de/)
+* general code metrics using [Scrutinizer](https://scrutinizer-ci.com/)
+
+### Unit tests
+
+The unit test files are in the `tests/` directory.
+
+To run the unit tests, just execute:
+
+```bash
+./tests/bin/run-unittests.sh
+```
+
+To write unit tests, find the relevant file that has similar tests already in them (they are named accordingly), and add the tests there. If there isn't a file that has tests for the functionality you've written, create a new file, and name it `test_<functionality>.php`. PHPUnit will pick up the file automatically. Refer to the [PHPUnit documentation](https://phpunit.de/manual/4.5/en/index.html) and existing files for how to write new tests.
+
+### Scrutinizer
+
+For the most part feedback from Scrutinizer will flag up potential issues with the submitted code. It will tell you whether anything has been introduced or fixed. It will also give you explanations of those.
+
+## Helpful tips for writing issues
+
+In order to best help you / verify bug reports, we need a few details in the issue to move forward.
```

Comment:
Maybe: "When submitting an issue on GitHub there are several things you should include to ensure we can verify the problem and have enough information to resolve it."